dali/kernels/slice/slice_flip_normalize_permute_pad_gpu.h

```diff
@@ -101,12 +102,26 @@ class SliceFlipNormalizePermutePadGpu {
     se.add<mm::memory_kind::pinned, detail::SampleDesc<Dims>>(num_samples);
     se.add<mm::memory_kind::device, detail::SampleDesc<Dims>>(num_samples);
 
+    int blocks_per_sm_;
+    CUDA_CALL(cudaOccupancyMaxActiveBlocksPerMultiprocessor(&blocks_per_sm_,
+      detail::SliceFlipNormalizePermutePadKernel<false, false, false, OutputType, InputType, Dims>,
+      kBlockDim,
+      0));
+
     block_count_ = 0;
+    auto number_of_blocks = GetSmCount() * blocks_per_sm_;
+    size_t all_sample_sizes = 0;
     for (auto &elem : args) {
       size_t sample_size = volume(elem.shape);
+      all_sample_sizes += sample_size;
       block_count_ += std::ceil(
         sample_size / static_cast<float>(kBlockSize));
     }
+    if (block_count_ % number_of_blocks != 0) {
+      block_count_ += (number_of_blocks - (block_count_ % number_of_blocks));
+    }
+    kBlockSize = div_ceil(all_sample_sizes, block_count_);
+    block_count_ = std::ceil(all_sample_sizes / static_cast<float>(kBlockSize));
```

Comment:
I think the bug that causes the tests to fail is in this line.

It'll be probably easiest to explain it by example. Each block can work on only one sample.  Let's say we have 10 samples, each of size 100, and kBlockSize is 1000. We need 10 blocks, because each block can work on only one sample. But `all_sample_sizes / kBlockSize` is `1`.

Calculating `block_count_` as a sum over the samples seems to do the trick:

```suggestion
    block_count_ = 0;
    for (auto &elem : args) {
      block_count_ += std::ceil(volume(elem.shape) / static_cast<float>(kBlockSize));
    }```